src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,1001 @@ static ColliderCylinderInit D_809D5C40 = {
+
+            return;
+        }
+    }
+}
+
+/**
+ * Creates a flash effect.
+ * Flash effects are spawned when the Arwing dies.
+ * Flash effects two components: 1) a billboard flash, and 2) a light effect on the ground.
+ */
+void EnClearTag_CreateFlashEffect(GlobalContext* globalCtx, Vec3f* position, f32 scale, f32 floorHeight,
+                                  Vec3f* floorTangent) {
+    s16 i;
+    EnClearTagEffect* effect;
+    GlobalContext* globalCtx2;
+
+    globalCtx2 = globalCtx;
+    effect = (EnClearTagEffect*)globalCtx2->specialEffects;
+
+    // Look for an available effect to allocate a flash effect to.
+    for (i = 0; i < CLEAR_TAG_EFFECT_MAX_COUNT; i++, effect++) {
+        if (effect->type == CLEAR_TAG_EFFECT_AVAILABLE) {
+            effect->type = CLEAR_TAG_EFFECT_FLASH;
+
+            effect->position = *position;
+            effect->velocity = sZeroVector;
+            effect->acceleration = sZeroVector;
+
+            effect->scale = 0.0f;
+            effect->maxScale = scale + scale;
+
+            effect->floorHeight = floorHeight;
+            effect->floorTangent = *floorTangent;
+
+            effect->primColor.a = 180.0f;
+
+            return;
+        }
+    }
+}
+
+/**
+ * EnClear_Tag destructor.
+ * This just destroys the collider.
+ */
+void EnClearTag_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnClearTag* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+/**
+ * EnClear_Tag constructor.
+ * This allocates a collider, initializes effects, and sets up ClearTag instance data.
+ */
+void EnClearTag_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnClearTag* this = THIS;
+    s32 defaultCutsceneTimer = 100;
+    s16 j, i;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+
+    // Initialize the Arwing laser.
+    if (this->actor.params == CLEAR_TAG_LASER) {
+        this->state = CLEAR_TAG_STATE_LASER;
+        this->work[0] = 70;
+        this->actor.speedXZ = 35.0f;
+        func_8002D908(&this->actor);
+        for (j = 0; j <= 0; j++) {
+            func_8002D7EC(&this->actor);
+        }
+        this->actor.scale.x = 0.4f;
+        this->actor.scale.y = 0.4f;
+        this->actor.scale.z = 2.0f;
+        this->actor.speedXZ = 70.0f;
+        this->actor.shape.rot.x = -this->actor.shape.rot.x;
+
+        func_8002D908(&this->actor);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sLaserCylinderInit);
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_SWORD_REFLECT_MG);
+        return;
+    }
+
+    // Initialize the Arwing.
+    this->actor.flags |= 1;
+    this->actor.targetMode = 5;
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sArwingCylinderInit);
+    this->actor.colChkInfo.health = 3;
+
+    // Update the Arwing to play the intro cutscene.
+    if (this->actor.params == CLEAR_TAG_CUTSCENE_ARWING) {
+        this->work[0] = 70;
+        this->work[1] = 250;
```

Comment:
if youre interested in doing more documentation, defines for the `work` indices would make things much more readable.
I also think we should consider a name other than work, but we dont have to figure that out now

Reply:
I think leaving it as `work` for now would be preferable. Then we can more easily replace all `work` variable names in the future.

Reply:
I did however make an enum.